packages/vite/src/node/utils.ts

```diff
@@ -18,6 +18,17 @@ export function slash(p: string): string {
   return p.replace(/\\/g, '/')
 }
 
+export async function usePromise<T, K>(
+  arr: T[],
+  processFn: (item: T) => Promise<K>
+): Promise<K[]> {
+  const queue = []
+  for (const item of arr) {
+    queue.push(processFn(item))
+  }
+  return Promise.all<K>(queue)
+}
```

Comment:
**suggestion (nitpick):** We may simplify this to e.g.

```suggestion
export async function usePromise<T, R>(
  arr: T[],
  processFn: (item: T) => Promise<R>
): Promise<R[]> {
  return Promise.all(arr.map((item) => processFn(item)))
}
```

---

But I'm not sure if we should use `Promise.allSettled` instead of `Promise.all`,
cause what if an output throws an error / rejects? Then all the promises gets cancelled :thinking:  
Previously it was at least `sequential` and therefore reproducible. But now there could be theoretically race conditions :thinking:

Reply:
once output generate is failed, just throw error,  don't have to let them all settled